pkg/genericcomp/fluentd.go

```diff
@@ -0,0 +1,189 @@
+  </record>
+</filter>
+<match **>
+  @type elasticsearch
+  host "#{ENV['ELASTICSEARCH_HOST']}"
+  port "#{ENV['ELASTICSEARCH_PORT']}"
+  user "#{ENV['ELASTICSEARCH_USER']}"
+  password "#{ENV['ELASTICSEARCH_PASSWORD']}"
+  index_name "#{ENV['APPLICATION_NAME']}"
+  scheme http
+  include_timestamp true
+  flush_interval 10s
+</match>
+`
+
+// CreateFluentdConfigMap creates the Fluentd configmap for a given generic application
+func CreateFluentdConfigMap(componentName string, namespace string, labels map[string]string) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getFluentdConfigMapName(componentName),
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Data: func() map[string]string {
+			var data map[string]string
+			data = make(map[string]string)
+			data[fluentdConf] = FluentdConfiguration
+			return data
+		}(),
+	}
+}
+
+// Create the Fluentd sidecar container
+func createFluentdContainer(bindingName string, componentName string) corev1.Container {
+	container := corev1.Container{
+		Name:            "fluentd",
+		Args:            []string{"-c", "/etc/fluent.conf"},
+		Image:           util.GetFluentdImage(),
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Env: []corev1.EnvVar{
+			{
+				Name:  "APPLICATION_NAME",
+				Value: componentName,
+			},
+			{
+				Name:  "FLUENTD_CONF",
+				Value: "fluentd.conf",
+			},
+			{
+				Name:  "FLUENT_ELASTICSEARCH_SED_DISABLE",
+				Value: "true",
+			},
+			{
+				Name:  "ELASTICSEARCH_HOST",
+				Value: fmt.Sprintf("vmi-%s-es-ingest.%s.svc.cluster.local", bindingName, constants.VerrazzanoNamespace),
+			},
+			{
+				Name:  "ELASTICSEARCH_PORT",
+				Value: "9200",
+			},
+			{
+				Name: "ELASTICSEARCH_USER",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: constants.VmiSecretName,
+						},
+						Key: "username",
+						Optional: func(opt bool) *bool {
+							return &opt
+						}(true),
+					},
+				},
+			},
+			{
+				Name: "ELASTICSEARCH_PASSWORD",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: constants.VmiSecretName,
+						},
+						Key: "password",
+						Optional: func(opt bool) *bool {
+							return &opt
+						}(true),
+					},
+				},
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				MountPath: "/fluentd/etc/fluentd.conf",
+				Name:      "fluentd-config-volume",
+				SubPath:   "fluentd.conf",
+				ReadOnly:  true,
+			},
+			{
+				MountPath: "/var/log",
+				Name:      "varlog",
+				ReadOnly:  true,
+			},
+			{
+				MountPath: "/u01/data/docker/containers",
+				Name:      "datadockercontainers",
+				ReadOnly:  true,
+			},
+		},
+	}
+
+	return container
+}
+
+// Create hostPath volumes for logs
+func createFluentdVolHostPaths() *[]corev1.Volume {
```

Comment:
Might be better to call this createFluentdVolumes or ..HostPathVolumes

Reply:
I will change the name of the function to createFluentdHostPathVolumes